dashboard/src/shared/schema.test.ts

```diff
@@ -0,0 +1,261 @@
+          },
+        },
+      },
+      result: {
+        username: {
+          name: "username",
+          path: "credentials.user",
+          value: "andres",
+        } as IBasicFormParam,
+      },
+    },
+    {
+      description: "should retrieve several params and ignore the ones not marked",
+      values: `
+# Application Credentials
+credentials:
+  admin:
+    user: andres
+    pass: myPassword
+
+# Number of Replicas
+replicas: 1
+
+# Service Type
+service: ClusterIP
+`,
+      schema: {
+        properties: {
+          credentials: {
+            type: "object",
+            properties: {
+              admin: {
+                type: "object",
+                properties: {
+                  user: { type: "string", form: "username" },
+                  pass: { type: "string", form: "password" },
+                },
+              },
+            },
+          },
+          replicas: { type: "number", form: "replicas" },
+          service: { type: "string" },
+        },
+      },
+      result: {
+        username: {
+          name: "username",
+          path: "credentials.admin.user",
+          value: "andres",
+        } as IBasicFormParam,
+        password: {
+          name: "password",
+          path: "credentials.admin.pass",
+          value: "myPassword",
+        } as IBasicFormParam,
+        replicas: {
+          name: "replicas",
+          path: "replicas",
+          value: 1,
+        } as IBasicFormParam,
+      },
+    },
+  ].forEach(t => {
+    it(t.description, () => {
+      expect(retrieveBasicFormParams(t.values, t.schema)).toEqual(t.result);
+    });
+  });
+});
+
+describe("getDocumentElem", () => {
+  [
+    {
+      description: "should get the root elem",
+      values: "foo: bar",
+      path: "foo",
+      result: "bar",
+    },
+    {
+      description: "should return a nested value",
+      values: "foo:\n  bar: foobar",
+      path: "foo.bar",
+      result: "foobar",
+    },
+    {
+      description: "should return a deeply nested value",
+      values: "foo:\n  bar:\n    foobar: barfoo",
+      path: "foo.bar.foobar",
+      result: "barfoo",
+    },
+    {
+      description: "should return a deeply nested value as an object",
+      values: "foo:\n  bar:\n    foobar: 1",
+      path: "foo.bar",
+      result: YAML.parseDocument("foo:\n  bar:\n    foobar: 1")
+        .get("foo")
+        .get("bar"),
+    },
+    {
+      description: "should ignore an invalid value",
+      values: "foo:\n  bar:\n    foobar: 1",
+      path: "nope",
+      result: undefined,
+    },
+    {
+      description: "should ignore an invalid value (nested)",
+      values: "foo:\n  bar:\n    foobar: 1",
+      path: "not.exists",
+      result: undefined,
+    },
+  ].forEach(t => {
+    it(t.description, () => {
+      const doc = YAML.parseDocument(t.values);
+      expect(getDocumentElem(doc, t.path)).toEqual(t.result);
+    });
+  });
+});
+
+describe("getDefaultValue", () => {
```

Comment:
Never mind, I see the test cases are identical because in the actual test for `getDocumentElem` you're parsing the YAML, where as in the tests for `getDefaultValue` you're passing in the YAML to be parsed.

See below, but I think we don't need `getDocumentElem` (as we can probably use lodash get for this), in which case, I'd rename `getDefaultValue` as `getYAMLValue` if that's what it is doing (taking a yaml string and getting the value at a path).

Reply:
I will rename it as `getValue` since it's actually the opposite of `setValue`